MDM-11596 add postinstall message for the DefinelyType

DrMabuse23 opened the pull request:
```bash
##########################################
# Please add the following typings manually:
# typings install dt~es6-collections dt~es6-promise dt~backbone dt~cordova/plugins/device dt~phonegap-plugin-push dt~websql dt~backbone-global dt~q dt~jquery dt~underscore dt~node --save --global
# typings install npm~lodash --save
# Thank you, the Relution Team.
##########################################
```

t-beckmann commented on bin/installInfo.js:
```diff
@@ -0,0 +1,8 @@
+'use strict';
+console.log('##########################################');
+console.log('# Please add the following typings manually:');
+console.log('# typings install dt~es6-collections dt~es6-promise dt~backbone dt~cordova/plugins/device dt~phonegap-plugin-push dt~websql dt~backbone-global dt~q dt~jquery dt~underscore dt~node --save --global');
```
The relution-sdk does not depend on es6-collections and es6-promise, so users of the library don't need to install these. (Your project probably uses these by itself, but this is not an SDK requirement, so just remove these two from the message.)

t-beckmann commented on bin/installInfo.js:
```diff
@@ -0,0 +1,8 @@
+console.log('##########################################');
+console.log('# Please add the following typings manually:');
+console.log('# typings install dt~es6-collections dt~es6-promise dt~backbone dt~cordova/plugins/device dt~phonegap-plugin-push dt~websql dt~backbone-global dt~q dt~jquery dt~underscore dt~node --save --global');
+console.log('# typings install npm~lodash --save')
```
The need for manual installation of typings should arise for global dependencies only. Doesn't it work without explicit installation of these? If not, something is wrong with the SDK setup itself I believe and we need to further investigate this. In this case, we should raise an issue and accept your MR as a workaround.

t-beckmann commented on package.json:
```diff
@@ -11,6 +11,9 @@
     "mvc"
   ],
+  "bin": {
+    "installInfo": "./bin/installInfo.js"
```
I need to try out on Windows, as you rely on the hash-bang logic of the unix shell here. Eventually we need to change this to "node ./bin/installInfo.js" to work cross-platform.

DrMabuse23 commented:
i think we have to change the typings install because there are always updates:
```bash
typings WARN deprecated 7/23/2016: "registry:npm/lodash#4.0.0+20160416211519" is deprecated (updated, replaced or removed)
typings WARN deprecated 7/23/2016: "registry:npm/chai#3.5.0+20160415060238" is deprecated (updated, replaced or removed)
typings WARN deprecated 7/23/2016: "registry:npm/sinon#1.16.0+20160309002336" is deprecated (updated, replaced or removed)
typings WARN deprecated 4/29/2016: "registry:npm/request#2.69.0+20160304121250" is deprecated (updated, replaced or removed)
typings WARN deprecated 7/20/2016: "registry:dt/mocha#2.2.5+20160317120654" is deprecated (updated, replaced or removed)
typings WARN deprecated 7/28/2016: "registry:dt/node#4.0.0+20160423143914" is deprecated (updated, replaced or removed)
typings WARN deprecated 6/13/2016: "registry:dt/q#0.0.0+20160417152954" is deprecated (updated, replaced or removed)
typings WARN deprecated 7/20/2016: "registry:dt/underscore#1.7.0+20160622050840" is deprecated (updated, replaced or removed)
```

t-beckmann commented:
Hm, this doesn't seem to work for me:

```
Thomas@MWAY-THOMAS /d/gofer-core/gofer-workspace/relution-sdk (feature/MDM-11596)
$ npm i

> relution-sdk@0.0.1 postinstall d:\gofer-core\gofer-workspace\relution-sdk
> installInfo && typings install

Der Befehl "installInfo" ist entweder falsch geschrieben oder
konnte nicht gefunden werden.

npm WARN relution-sdk@0.0.1 No repository field.
npm WARN relution-sdk@0.0.1 license should be a valid SPDX license expression
npm ERR! Windows_NT 10.0.10586
npm ERR! argv "c:\\Program Files\\nodejs\\node.exe" "c:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "i"
npm ERR! node v4.4.4
npm ERR! npm  v3.9.5
npm ERR! code ELIFECYCLE
npm ERR! relution-sdk@0.0.1 postinstall: `installInfo && typings install`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the relution-sdk@0.0.1 postinstall script 'installInfo && typings install'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the relution-sdk package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     installInfo && typings install
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs relution-sdk
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls relution-sdk
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     d:\gofer-core\gofer-workspace\relution-sdk\npm-debug.log
```

t-beckmann commented:
It does work however when doing the following in package.json:

```
    "postinstall": "node bin/installInfo.js && typings install",
```

I'm pretty sure this is a windows thing.

t-beckmann commented:
https://github.com/relution-io/relution-sdk/pull/13 solves this issue without users requiring to do typings installations.
